src/steps/lead-field-equals.ts

```diff
@@ -41,17 +42,28 @@ export class LeadFieldEquals extends BaseStep implements StepInterface {
       filter: `startswith(emailaddress1, '${stepData.email}')`,
       count: true,
     };
+    let actualValue;
 
     try {
       const records = await this.client.retrieveMultiple(request);
       const result = records.find((lead: any) => lead['emailaddress1'] === email);
-      if (this.compare(operator, result[field], expectedValue)) {
+
+      if (!result) {
+        return this.error('No Lead was found with email %s', [email]);
+      } else {
+        actualValue = result[field];
+        if (isDate(result[field])) {
+          actualValue = JSON.stringify(result[field]).replace(/\"/g, '');
```

Comment:
Oh neat!  How kind of the API client to return date fields as native JS `Date` objects.

- [ ] Recommend `result[field].toISOString()` instead of the stringify here.  See [Date.prototype.toISOString](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toISOString) docs.

Reply:
Actually spent a lot of time researching about dateTime or why its displayed as purple on console log and changing display on toString() yesterday. This info helps a ton hahaha thanks so much :)